src/lib/io/sol-memmap-storage-impl-zephyr.c

```diff
@@ -0,0 +1,412 @@
+        memcpy(back_mem + mod, blob->mem, size);
+
+        r = flash_write_protection_set(map_internal->flash_dev, false);
+        SOL_INT_CHECK_GOTO(r, < 0, error);
+
+        for (cnt = 0; cnt < back_sz; cnt += map_internal->max_rw_sz) {
+            r = flash_write(map_internal->flash_dev, offset - mod + cnt,
+                back_mem + cnt, map_internal->max_rw_sz);
+            if (r < 0) {
+                SOL_WRN("Flash write failed");
+                goto error;
+            }
+            r = flash_write_protection_set(map_internal->flash_dev, false);
+            SOL_INT_CHECK_GOTO(r, < 0, error);
+        }
+    }
+
+    if (cb)
+        cb((void *)data, name, blob, r);
+
+    sol_blob_unref(blob);
+
+    return r;
+
+error:
+    SOL_DBG("Error writing to memmap: %s", sol_util_strerrora(-r));
+
+    if (cb)
+        cb((void *)data, name, blob, r);
+
+    sol_blob_unref(blob);
+
+    return r;
+}
+
+bool
+sol_memmap_impl_perform_pending_writes(void *data)
+{
+    struct map_internal_zephyr *map_internal = data;
+    struct pending_write_data *pending;
+    struct sol_vector tmp_vector;
+    int i;
+
+    //sol-memmap-storage.c only checks the pointer to know what to do
+    //on the next round
+    map_internal->base.timeout = NULL;
+
+    tmp_vector = map_internal->base.pending_writes;
+    sol_vector_init(&map_internal->base.pending_writes,
+        sizeof(struct pending_write_data));
+
+    SOL_VECTOR_FOREACH_IDX (&tmp_vector, pending, i) {
+        sol_memmap_impl_write_raw((struct map_internal *)map_internal,
+            pending->name, pending->entry, pending->mask, pending->blob,
+            pending->cb, pending->data);
+        sol_blob_unref(pending->blob);
+    }
+    sol_vector_clear(&tmp_vector);
+
+    SOL_DBG("Performed pending writes");
+
+    return false;
+}
+
+int
+sol_memmap_impl_init(void)
+{
+    return 0;
+}
+
+#define DRIVER_NAME_IDX 0
+#define MIN_ERASE_SZ_IDX 1
+#define MAX_RW_SZ_IDX 2
+#define MEM_OFFSET_IDX 3
+
+static int
+resolve_map_path(const struct sol_memmap_map *map,
+    struct map_internal_zephyr *map_internal,
+    char **driver_name)
+{
+    char *min_erase_size_s = NULL, *max_rw_size_s = NULL, *mem_offset_s = NULL,
+    *end_ptr;
+    unsigned int min_erase_size, max_rw_size;
+    struct sol_vector instructions;
+    struct sol_str_slice command;
+    int ret = 0;
+
+    SOL_NULL_CHECK(driver_name, -EINVAL);
+    *driver_name = NULL;
+
+    command = sol_str_slice_from_str(map->path);
+
+    instructions = sol_str_slice_split(command, ",", 4);
+    if (instructions.len < 4)
+        goto err_parse;
+
+    *driver_name = sol_str_slice_to_string(
+        *(const struct sol_str_slice *)sol_vector_get(&instructions,
+        DRIVER_NAME_IDX));
+    SOL_NULL_CHECK_GOTO(*driver_name, err_parse);
+
+    min_erase_size_s = sol_str_slice_to_string(
+        *(const struct sol_str_slice *)sol_vector_get(&instructions,
+        MIN_ERASE_SZ_IDX));
+    SOL_NULL_CHECK_GOTO(*min_erase_size_s, err_parse);
```

Comment:
This kind of 'instruction mapping' code is appearing more and more. Couldn't we unify them? On Linux you changed it to `sscanf`, just to use `sol_str_slice` here again...